docs/apache.md

```diff
@@ -26,3 +26,134 @@ RewriteRule /wetty/socket.io/(.*) ws://localhost:3000/wetty/socket.io/$1 [P,L]
+
+                # this propagates to apache2 (and thus to access log) the proper user id, and not
+                # the transiant nameid that is taken by default
+                # it has no impact on the backend as we propagate identify via remote-user header there
+                MellonUser "NameID"
+
+                MellonEndpointPath /mellon/
+                MellonSPMetadataFile /etc/apache2/saml2/foo.xml
+                MellonSPPrivateKeyFile /etc/apache2/saml2/foo.key
+                MellonSPCertFile /etc/apache2/saml2/foo.cert
+                MellonIdPMetadataFile /etc/apache2/saml2/idp.xml
+
+                # the identity propagated to wetty (as HTTP header 'remote-user: xxxxx')
+                # is retrieved from SAMLResponse NameID attribute
+                RequestHeader set remote-user %{MELLON_NAMEID}e
+        </Location>
+
+        <Location /wetty>
+                AuthType Mellon
+                MellonEnable auth
+                Require valid-user
+        </Location>
+
+        # security hazard for switching between users, disabled if remote-user set as recent github commit
+        # but not yet published via yarn, so we put here a double security belt
+        <Location /wetty/ssh/>
+            Deny from all
+        </Location>
+</VirtualHost>
+```
+
+### Auto login
+
+If you want to have a seamless login by trusting your IdP for
+authentication, you can create password-less users on the wetty
+platform and have them trust an SSH key used by the NodeJS, owned by
+the dedicated wetty OS user.
+
+Wetty instanciation with proper parameters, especially the SSH private
+key is done via the following systemd service
+`/etc/systemd/system/wetty.service`:
+
+```
+[Unit]
+Description=Wetty Web Terminal
+After=network.target
+
+[Service]
+User=wetty
+Type=simple
+WorkingDirectory=/home/wetty/.config/yarn/global/node_modules/wetty/
+ExecStart=/usr/bin/node . -p 3000 --host 127.0.0.1 --ssh-key /home/wetty/.ssh/wetty --ssh-auth publickey --force-ssh --title "Foo bar terminal services"
+TimeoutStopSec=20
+KillMode=mixed
+Restart=always
+RestartSec=2
+
+[Install]
+WantedBy=multi-user.target
+```
+
+For your new users to be automically trusting this SSH key when
+provisionning, you may add the pubkey to
+`/etc/skel/.ssh/authorized_keys`.
+
+
+### Security precautions
+
+You probably don't want local users to impersonate each other, for that you need to make sure that:
+
+1. NodeJS is listenning only to localhost: provided by `wetty.service`
+2. **Only** the apache2 process can join the wetty port. Else local users
+   will be able to connect and forge a `remote-user` header: provided
+   by `iptables -A OUTPUT -o lo -p tcp --dport 3000 -m owner \!
+   --uid-owner www-data -j DROP`
+3. Validate your wetty version does not allow access to `/wetty/ssh/`
+   else again you will be able to impersonnate anyone: provided by
+   either:
+   1. a recent wetty commit disabling it in case of `remote-user`
```

Comment:
say version 2.0.3 and beyond

Reply:
new changes implement this (I'm wondering If I need to do something to have it included in the current PR)

Reply:
nope your good